client/components/replacePost/index.jsx

```diff
@@ -0,0 +1,100 @@
+/* eslint-disable no-underscore-dangle */
+/* global React */
+
+import PostSelector from '../postSelector';
+
+const {
+  apiFetch,
+  components: {
+    Spinner,
+  },
+  data: {
+    select,
+    useDispatch,
+  },
+  element: {
+    useState,
+    useEffect,
+  },
+  i18n: {
+    __,
+  },
+} = wp;
+
+/**
+ * Slotfill to add clone & replace controls to the sidebar.
+ */
+const ReplacePost = () => {
+  const { editPost } = useDispatch('core/editor');
+  const currentPost = select('core/editor').getCurrentPost();
+  const postType = select('core/editor').getCurrentPostType();
+  const meta = select('core/editor').getEditedPostAttribute('meta') || {};
+  const [replacePostId, setReplacePostId] = useState(meta._cr_replace_post_id);
+  const [replacePost, setReplacePost] = useState(false);
+  const selected = replacePost ? [replacePost] : [];
+
+  const fetchPost = async (postId) => {
+    const post = await apiFetch({ path: `/wp/v2/${postType}/${postId}` });
+    console.log(post);
+    setReplacePost({
+      id: post.id,
+      title: post.title.rendered,
+    });
+  };
+
+  useEffect(() => {
+    editPost({
+      meta: {
+        ...meta,
+        _cr_replace_post_id: replacePostId.toString(),
+      },
+    });
+  }, [replacePostId]);
+
+  useEffect(() => {
+    if (replacePostId) {
+      fetchPost(replacePostId);
+    }
+  }, []);
+
+  /**
+   * We only are interested in draft posts here.
```

Comment:
This seems like the opposite of the previous check - is that intentional? Different contexts?

Reply:
You can only set a draft post to replace a published post so this is intentional.

Reply:
👍